extension/src/test/suite/project.test.ts

```diff
@@ -49,15 +54,196 @@ suite("Project related Tests", function () {
+			ProjectTreeProvider.instance().updateData(null);
+			try{
+				await excludeFromProject(rootNode.sourceFiles[0]);
+			}
+			catch (err) {
+				expect(err.startsWith("A project must be open")).to.be.true;
+			}
+		});
+	});
+
+	suite("Project save project Test", function(){
+		test("Successed to save project", async function(){
+			let destprojpath = path.join(prefixpath + "project_test_file_add.prj");
+			fs.copyFileSync(projpath, destprojpath);		
+			let ret = vscode.Uri.file(destprojpath);
+			let rootNode= OpenProjectFile(ret);
+			ProjectTreeProvider.instance().updateData(rootNode);
+			let fileName= path.join(prefixpath + "pdfMarkups.lsp");
+			try{
+				addLispFileNode2ProjectTree(rootNode, fileName, null);
+				await SaveProject(true);
+				OpenProjectFile(ret);
+				expect(rootNode.sourceFiles.length).to.be.equals(2);
+			}
+			catch (err) {
+				expect(err).to.be.equals("");
+			}
+			fs.removeSync(destprojpath);
+		});	
+		
+		test("No project to save", async function(){
+			ProjectTreeProvider.instance().updateData(null);
+			try{
+				let msg = SaveProject(true);
+				msg
+				.then(response => {
+					return;
+				})
+				.catch(error => {
+					expect(error).to.be.equals("No project to save.");
+				})
+			}
+			catch (err) {
+				console.log(err);
+				expect(err).to.be.equals("");
+			}
+		});	
+	});
+
+	suite("Project refresh Test", function(){
+		test("Successed to refresh project", function(){
+			try {
+				RefreshProject();
+			}
+			catch (err) {
+				console.log(err);
+				expect(err).to.be.equals("");
+			}
+		});
+
+		test("file with same name", function(){
+			let rootNode= OpenProjectFile(ret);
+			let fileName = "test_remove.lsp";
+			try {
+				hasFileWithSameName(fileName, rootNode);
+			}
+			catch (err) {
+				console.log(err);
+				expect(err).to.be.equals("");
+			}
+		});
+	});
+
+	suite("Project getTreeItemTitle Test", function(){
+		test("Successed to getTreeItemTitle ptoject", function(){
+			let treeNode = new vscode.TreeItem(rootNode.getDisplayText());
+			try {
+				getTreeItemTitle(treeNode);
+			}
+			catch (err) {
+				console.log(err);
+				expect(err).to.be.equals("");
+			}
+			
+		});
+	});
+
+	suite("Project open lisp file Test", function(){
+		test("Successed to open lisp file", async function(){
+			let clickedTreeItem = rootNode.sourceFiles[0];
+			try {
+				await openLspFile(clickedTreeItem);
+			}
+			catch (err) {
+				console.log(err);
+				expect(err).to.be.equals("");
+			}
+		});
+
+		test("Failed to open lisp file", async function(){
+			let projpath = path.join(prefixpath + "project_no_lisp_file.prj");	
+			let ret = vscode.Uri.file(projpath);
+			let rootNode= OpenProjectFile(ret);
+			let clickedTreeItem = rootNode.sourceFiles[0];
+			try{
+				await openLspFile(clickedTreeItem);
+			}
+			catch (err) {
+				expect(err.startsWith("File doesn't exist: ")).to.be.true;
+			}
+		});
+	});
+<<<<<<< HEAD
+});
+=======
 });
+>>>>>>> wip/ss/add_lisp_to_project
```

Comment:
```suggestion
});
```

Reply:
@shuang6
I updated your branch and resolved the conflicts, please check if the latest code is correct.